api/src/main/java/uk/gov/dvsa/mot/app/MotrRequestHandler.java

```diff
@@ -0,0 +1,252 @@
+
+    private MotrReadService motrReadService;
+    private VehicleReadService vehicleReadService;
+    private HgvVehicleProvider hgvVehicleProvider;
+    private String awsRequestId;
+
+    public MotrRequestHandler() {
+        super();
+    }
+
+    public MotrRequestHandler(boolean injectSelf) {
+        super(injectSelf);
+    }
+
+    @Inject
+    public void setVehicleReadService(VehicleReadService vehicleReadService) {
+        this.vehicleReadService = vehicleReadService;
+    }
+
+    @Inject
+    public void setHgvVehicleProvider(HgvVehicleProvider hgvVehicleProvider) {
+        this.hgvVehicleProvider = hgvVehicleProvider;
+    }
+
+    @Inject
+    public void setMotrReadService(MotrReadService motrReadService) {
+        this.motrReadService = motrReadService;
+    }
+
+    @GET
+    @Path("motr/v2/search/registration/{registration}")
+    @Produces("application/json")
+    public Response getVehicle(@PathParam("registration") String registration,
+                               ContainerRequestContext requestContext) throws TradeException {
+        try {
+
+            ApiGatewayRequestContext context = (ApiGatewayRequestContext) requestContext.getProperty(
+                    RequestReader.API_GATEWAY_CONTEXT_PROPERTY);
+            if (context != null) {
+                awsRequestId = context.getRequestId();
+            }
+
+            logger.info(String.format("Entering MotrRequestHandler, awsRequestId: %s", awsRequestId));
+
+            if (Strings.isNullOrEmpty(registration)) {
+                logger.error("Registration param is missing");
+                throw new BadRequestException("Registration param is missing", awsRequestId);
+            }
+
+            Optional<MotrResponse> motVehiclesOptional = getMotVehicle(registration);
+
+            if (motVehiclesOptional.isPresent()) {
+
+                return Response.ok(setResponseMotVehicle(motVehiclesOptional.get())).build();
+            }
+
+            Optional<DvlaVehicle> dvlaVehicleOptional = getDvlaVehicle(registration);
+
+            if (dvlaVehicleOptional.isPresent()) {
+                DvlaVehicle dvlaVehicle = dvlaVehicleOptional.get();
+
+                Optional<Vehicle> hgvPsvVehicleOptional = getHgvPsvVehicle(registration);
+
+                if (!hgvPsvVehicleOptional.isPresent()) {
+                    logger.error("No HGV/PSV vehicle retrieved");
+                    throw new InvalidResourceException("No HGV/PSV vehicle retrieved", awsRequestId);
+                }
+
+                HgvPsvVehicle hgvPsvVehicle = setResponseHgvVehicle(hgvPsvVehicleOptional.get(), dvlaVehicle);
+
+                return Response.ok(hgvPsvVehicle).build();
+            } else {
+                logger.error("No MOT Test or DVLA vehicle found for registration.");
+                throw new InvalidResourceException("No MOT Test or DVLA vehicle found for registration " + registration,
```

Comment:
InvalidResourceException - what is the effect of throwing that exception ? What response will be produced ?

Reply:
Http: 404 not found with for example:

{ "httpStatus": "404", "errorMessage": "No MOT Test or DVLA vehicle found for registration EXTODAYY", "awsRequestId": "xxx"}